infra/src/main/kotlin/fi/fta/geoviite/infra/authorization/UserData.kt

```diff
@@ -26,32 +23,36 @@ data class Privilege(val code: AuthCode) : GrantedAuthority {
     override fun getAuthority(): String = code.toString()
 }
 
-val userNameLength = 3..300
-
 data class UserName private constructor(private val value: String)
     : Comparable<UserName>, CharSequence by value {
     companion object {
+        val allowedLength = 3..300
+        const val ALLOWED_CHARACTERS = "A-Za-z0-9_\\-"
+        val sanitizer = StringSanitizer(UserName::class, ALLOWED_CHARACTERS, allowedLength)
+
         @JvmStatic
         @JsonCreator
-        fun of(name: String) = UserName(removeLogUnsafe(name))
+        fun of(name: String) = UserName(sanitizer.sanitize(name))
     }
-    init { assertLength<UserName>(value, userNameLength) }
+    init { sanitizer.assertSanitized(value) }
 
     @JsonValue
     override fun toString(): String = value
     override fun compareTo(other: UserName): Int = value.compareTo(other.value)
 }
 
-val authNameLength = 1..300
-
 data class AuthName private constructor(private val value: String)
     : Comparable<AuthName>, CharSequence by value {
     companion object {
+        val allowedLength = 1..300
```

Comment:
Tämä kuulostaa kanssa aika pitkältä, mutta toisaalta jos/kun nämä luetaan Projektivelhosta, niin ehkä ihan hyvä näin